### PR TITLE
Some more cleanups in logic

### PR DESCRIPTION
**Pull request by oscargus**

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/formatter/bibtexfields/PageNumbersFormatter.java [hunk 1]
```diff
@@ -18,6 +18,9 @@
 
     private static final Pattern PAGES_PATTERN = Pattern.compile("\\A(\\d+)-{1,2}(\\d+)\\Z");
 
+    private static final String REJECT_LITERALS = "[^0-9,\\-\\+]";
+    private static final String REPLACE = "$1--$2";
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/formatter/bibtexfields/PageNumbersFormatter.java]: Replace is not very descriptive. What about PagePattern or if you want ReplacePattern?

File: src/main/java/net/sf/jabref/logic/formatter/bibtexfields/SuperscriptFormatter.java [hunk 2]
```diff
@@ -28,6 +28,8 @@
     private static final Pattern PATTERN = Pattern.compile("\\b(\\d+)(st|nd|rd|th)\\b",
             Pattern.CASE_INSENSITIVE | Pattern.MULTILINE);
 
+    // adds superscript tag
+    private static final String REPLACE = "$1\\\\textsuperscript{$2}";
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/formatter/bibtexfields/SuperscriptFormatter.java]: Same as above. Also the comment would be better for line 63 I think

File: src/main/java/net/sf/jabref/logic/search/describer/GrammarBasedSearchRuleDescriber.java [hunk 3]
```diff
@@ -109,8 +109,8 @@ public String visitComparison(SearchParser.ComparisonContext context) {
             }
 
         }.visit(parseTree));
-        stringBuilder.append(". ");
-        stringBuilder.append(caseSensitive ? Localization.lang("The search is case sensitive.") :
+        stringBuilder.append(". ").append(caseSensitive ? Localization
+                .lang("The search is case sensitive.") :
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/search/describer/GrammarBasedSearchRuleDescriber.java]: I actually found the previous code more readable.

**Comment by tobiasdiez** [comment]
:+1: 
Good to go, except some minor remarks.

**Comment by oscargus** [comment]
Thanks! I've updated accordingly.

**Pull request merged**